### PR TITLE
Defibrillator Fixes + Code Cleanup

### PR DESCRIPTION
**Pull request by SpaceDragon00**
## About The Pull Request

Cleans up some of the defib code by reducing repeated code blocks, renaming certain procs, reducing unnecessary if statements, and adding flexability to certain aspects, like cooldown time. Additionally fixes some issues with the code, like visible messages not showing up for mobs who are wearing the source and shock_touching not using its provided damage values.

I have defined the combat paddles' emp_act change as a "Fix" because the comment appended to the combat variable, as well as the lack of sprites/clarity for emped syndicate defibs, made me feel this was unintentional. However, if this is seen as intentional, I will revert those changes to keep this a purely fix/cleanup-based PR

## Why It's Good For The Game

Cleaner code and fewer bugs

## Changelog
:cl:
fix: Combat defibs no longer enable their safeties when being EMPed
fix: Visible message sends its message to mobs that are also its location
fix: shock_touching (Renamed to shock_pulling) now actually uses the damage value provided
code: Removed useless mob/user input for cooldowncheck
code: Added recharge_time variable to paddles for variable charge times
code: Added can_hurt, do_success, and do_cancel to reduce code duplication
/:cl:

### CODE REVIEW

**Comment by Colovorat** [comment]
>fix: Combat defibs no longer enable their safeties when being EMPed
So when i did this one, it was balance change that was not merged...
Huh...

**Comment by SpaceDragon00** [comment]
> 
> 
> > fix: Combat defibs no longer enable their safeties when being EMPed
> 
> So when i did this one, it was balance change that was not merged...
> Huh...

I haven't gotten confirmation on whether or not it's intentional yet, that's just my initial feeling. I'll change it back if such is requested

File: code/game/objects/items/defib.dm [hunk 1]
```diff
@@ -468,17 +473,43 @@
...
 	do_help(H, user)
 
-/obj/item/shockpaddles/proc/shock_touching(dmg, mob/H)
+/// Returns TRUE when the paddles can do harmful actions, and false if they cant (If the safety is off, or if they're combat paddles)
+/obj/item/shockpaddles/proc/can_hurt()
+	if(combat)
+		return TRUE
+	if(req_defib && !defib.safety)
+		return TRUE
+	return FALSE
```
Comment (Rohesie) [inline code/game/objects/items/defib.dm:482]: I'm generally not a big fan of `is_thing()` or `can_do_thing()` procs that just check several conditions.
They tend to grow, be redefined for extra exceptions, and then make producing events-oriented code way harder.

My suggestion is to either make `can_hurt` a variable that changes based on the changes of the other variables (through setters), or alternatively just to turn this into a macro, which has lower chances of being expanded into:
`#define SHOCKPADDLES_CAN_HURT (combat || (req_defib && !defib.safety))`

Comment (SpaceDragon00) [reply code/game/objects/items/defib.dm:482]: Good point, I'll turn it into a define

**Comment by Rohesie** [comment]
Should be good after the merge conflict is solved.

File: code/game/objects/items/defib.dm [hunk 2]
```diff
@@ -161,7 +162,11 @@
 		deductcharge(1000 / severity)
 	if (. & EMP_PROTECT_SELF)
 		return
-	if(safety)
+
+	if(combat) // Elite agents do not subscribe to your notion of "Safety"
+		visible_message("<span class='notice'>[src] beeps: Safety protocols nonexistent!</span>")
```
Comment (Rohesie) [inline code/game/objects/items/defib.dm:167]: This needs to use the new procs.

**Review by Rohesie — CHANGES_REQUESTED** [review]

File: code/game/objects/items/defib.dm (hunk 2 above)
Comment (SpaceDragon00) [reply code/game/objects/items/defib.dm]: Sorry, silly mistake on my end

**Pull request merged**